Apps/g.Tec/g.NEEDaccess/src/mainwindow.cpp

```diff
@@ -0,0 +1,557 @@
+
+	strcpy(this->m_hostEndpoint.IpAddress, ui->lineEdit_serverip->text().toLatin1().data());
+	this->m_hostEndpoint.Port = ui->serverPortSpinBox->value();
+	this->m_localEndpoint.Port = ui->clientPortSpinBox->value();	
+
+	GDS_DEVICE_CONNECTION_INFO* connected_devices = NULL;
+	size_t count_daq_units = 0;
+	handleResult("GDS_GetConnectedDevices",
+		GDS_GetConnectedDevices(m_hostEndpoint, m_localEndpoint, &connected_devices, &count_daq_units));
+
+	std::vector<GDS_DeviceInfo> new_devices;
+	uint32_t count_devices = 0;
+	for (size_t i = 0; i < count_daq_units; i++)
+	{
+		// Each daq unit operates a different number of devices. The second "for" loop 
+		// addresses the devices attached to the daq unit.
+		for (size_t j = 0; j < connected_devices[i].ConnectedDevicesLength; j++)
+		{
+			GDS_DeviceInfo new_info;
+			new_info.index = count_devices;
+			new_info.inUse = bool(connected_devices[i].InUse) == true;
+			new_info.name = connected_devices[i].ConnectedDevices[j].Name;
+			new_info.type = connected_devices[i].ConnectedDevices[j].DeviceType;
+			new_devices.push_back(new_info);
+			count_devices++;
+		}
+	}
+	success &= handleResult("GDS_FreeConnectedDevicesList",
+		GDS_FreeConnectedDevicesList(&connected_devices, count_daq_units));
+
+	if (new_devices.size() > 0)
+	{
+		ui->availableListWidget->clear();
+		QStringList newDeviceNames;
+		for (auto it = new_devices.begin(); it != new_devices.end(); ++it) {
+			ui->availableListWidget->addItem(it->name);
+			// TODO: Use other GDS_DeviceInfo fields
+		}
+		ui->availableListWidget->setCurrentRow(0);
+	}
+	ui->scanPushButton->setText("Scan");
+	ui->scanPushButton->setDisabled(false);
+}
+
+
+void MainWindow::on_availableListWidget_itemSelectionChanged()
+{
+	ui->connectPushButton->setDisabled(false);
+}
+
+
+void MainWindow::on_connectPushButton_clicked()
+{
+	ui->goPushButton->setDisabled(true);
+	QString pbtext = ui->connectPushButton->text();
+	if (pbtext == "Connect")
+	{
+		QString deviceName = ui->availableListWidget->currentItem()->text();
+		char(*device_names)[DEVICE_NAME_LENGTH_MAX] = new char[1][DEVICE_NAME_LENGTH_MAX];
+		std::strcpy(device_names[0], deviceName.toLatin1().data());
+		BOOL is_creator = FALSE;
+		if (handleResult("GDS_Connect",
+			GDS_Connect(m_hostEndpoint, m_localEndpoint, device_names, 1, FALSE, &m_connectionHandle, &is_creator)))
+		{
+			ui->connectPushButton->setText("Disconnect");
+			ui->goPushButton->setDisabled(false);
+			enable_config_elements(false);
+			m_bConnected = true;
+			m_devInfo.is_creator = bool(is_creator);
+
+			if (!m_devInfo.is_creator)
+			{
+				m_devInfo.is_creator = handleResult("GDS_BecomeCreator",
+					GDS_BecomeCreator(m_connectionHandle));
+				if (!m_devInfo.is_creator)
+					statusBar()->showMessage("Device is first connected by another application. Checkboxes will be ignored.");
+			}
+			// checkboxes->setDisabled(!m_devInfo.is_creator)
+			ui->CARCheckBox->setDisabled(!m_devInfo.is_creator);
```

Comment:
I think there's a way to create a group in the designer and disabling it, so new checkboxes don't have to be added here.